common/djangoapps/student/models.py

```diff
@@ -3461,6 +3463,27 @@ def should_celebrate_first_section(enrollment):
         except CourseEnrollmentCelebration.DoesNotExist:
             return False
 
+    @staticmethod
+    def should_celebrate_weekly_goal(enrollment):
+        """ Returns the celebration value for weekly_goal with appropriate fallback if it doesn't exist """
+        if not enrollment:
+            return False
+        try:
+            if enrollment.celebration.celebrate_weekly_goal:
```

Comment:
Maybe move this check up to the initial early exit check? Might make "when this method really does something" clearer.